lang/py/src/avro/io.py

```diff
@@ -119,11 +120,27 @@ def validate(expected_schema, datum):
       return isinstance(datum, Decimal)
     return isinstance(datum, str)
   elif schema_type == 'int':
-    return ((isinstance(datum, int) or isinstance(datum, long)) 
-            and INT_MIN_VALUE <= datum <= INT_MAX_VALUE)
+    if hasattr(expected_schema, 'logical_type'):
+      if expected_schema.logical_type == 'date':
+        return isinstance(datum, datetime.date)
+      elif expected_schema.logical_type == 'time-millis':
+        return isinstance(datum, datetime.time)
+      else:
+        return False
+    else:
+      return ((isinstance(datum, int) or isinstance(datum, long))
+              and INT_MIN_VALUE <= datum <= INT_MAX_VALUE)
   elif schema_type == 'long':
-    return ((isinstance(datum, int) or isinstance(datum, long)) 
-            and LONG_MIN_VALUE <= datum <= LONG_MAX_VALUE)
+    if hasattr(expected_schema, 'logical_type'):
+      if expected_schema.logical_type == 'time-micros':
+        return isinstance(datum, datetime.time)
```

Comment:
Since `time_millis` and `time_micros` are checking `isinstance(datum, datetime.time)` we can do the same as in #137